taichi/analysis/clone.cpp

```diff
@@ -112,7 +112,7 @@ class IRCloner : public IRVisitor {
 
   static std::unique_ptr<IRNode> run(IRNode *root, Kernel *kernel) {
     if (kernel == nullptr) {
-      kernel = &get_current_program().get_current_kernel();
+      kernel = root->get_kernel()->program.current_kernel;
```

Comment:
This looks redundant, I think `root->get_kernel()` would do the job

Reply:
Right! `get_current_kernel` is a bad design and it introduces quite a few issues when I was implementing parallel compilation: multiple threads might fight for `Program::current_kernel`.